### ISSUE TITLE
Entering first character into a note, switches notes. 

### ISSUE THREAD

**Issue by rcprcp** [laurent22/joplin#10236]
### Operating system

macOS

### Joplin version

3.0.2

### Desktop version info

Joplin version
Joplin 3.0.2 (prod, darwin)

Sync Version: 3
Profile Version: 46
Keychain Supported: Yes

### Current behaviour

(this was also mentioned in [Issue 10194 ](https://github.com/laurent22/joplin/issues/10194), but it may be a different bug.) 

Using version 3.0.2, I have noticed that this sequence of steps results in incorrect behavior. Upon entering the first character, the current note changes to a different note. 

- Enter something into the search field.
- Ensure that the are multiple notes listed in the results.
- Select a note from the results.
- Click into the note's body; type just one character into the selected note. (The issue does not seem to happen if you update the note's title.) The UI updates, and you're viewing a different note. It seems to be one from the list, but not the originally selected one.

### Expected behaviour

After searching, you should be able to select one of the results, then click into a note body and edit the note.

**Comment by Marain888** [comment]
1. I am also experiencing this issue. 
2. I am using Desktop version Joplin 3.0.2 (prod, win32)
Sync Version: 3
Profile Version: 46
Keychain Supported: Yes
Revision: a59ad20
Backup: 1.4.0

3. Description: When I am in the All Notes view, sorted by most recently updated, I select any note in the list. I type any character in the editing panel for the note and suddenly I am in the next note in the list and the note I wanted to edit has disappeared from the list. I was able to see it in the list again by going anywhere else and coming back to the All Notes view, at which point, the note I tried to edit is at the top of the list and includes the character I entered. This only seems to happen in the All Notes view. If I go to the folder where the note actually resides, I am able to edit without any problem. 
4. This is a new behavior that I had never experienced before this latest release.
5. I enabled the crash reporting feature. 
6. I setup debug and have copied the console text (log is in next message) that resulted when I reproduced the error.
7. Screenshots:
![1](https://github.com/laurent22/joplin/assets/165838447/8315b91b-6ddc-440d-9134-4b0b1f74ef1e)
![2](https://github.com/laurent22/joplin/assets/165838447/2c057312-55f5-4e41-8ead-84c6a10ff03d)
![3](https://github.com/laurent22/joplin/assets/165838447/380e450a-6349-498d-82ee-2514e677ea44)
![4](https://github.com/laurent22/joplin/assets/165838447/4ff7d09f-049d-469d-bb09-240b3929b835)
![5](https://github.com/laurent22/joplin/assets/165838447/ab4d80a7-dcbb-49c6-9a44-2bad3cb55427)

**Comment by Marain888** [comment]
Debug log.
[-1712088635692.log](https://github.com/laurent22/joplin/files/14843210/-1712088635692.log)

**Comment by Marain888** [comment]
This bug still exists in 

Joplin 3.0.3 (prod, win32)
Client ID: ec934152e8ab441e8d1466e69798f9ff
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes
Revision: e84ea8a
Backup: 1.4.1

**Comment by Dricc123** [comment]
This bug still exists in
Joplin 3.0.6 (prod, win32)
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes
Revision: 18b9f5c

**Comment by Marain888** [comment]
**UPDATED: 2024-05-23** @Dricc123 - It seems to be okay for me now **in All Notes**<s>, but maybe my issue was different from yours. In my case, I could only edit one character in any Note in All Notes, and it would then "disappear" from the All Notes list (I would be able to see it again if I clicked on some other folder and then went back to All Notes). I could only edit a Note in its Home folder. This behavior seems to be gone now.</s> **However, in a List of Notes generated by a Search or with a Tag, this behaviour still occurs.**

Joplin 3.0.6 (prod, win32) 
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes
Revision: 18b9f5c

**Joplin 3.0.8 (prod, win32)
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes
Revision: 826006c
Backup: 1.4.1
Search & Replace: 2.2.0**

Marian

**Comment by Dricc123** [comment]
@Marain888 indeed you are referring to another issue that is solved for me as well. The present issue is described in details above.

**Comment by nonobio** [comment]
https://github.com/laurent22/joplin/issues/10388#issuecomment-2114741089

**Comment by Julrob199** [comment]
Hope this gets fixed soon! This bug still exists in 3.0.7, it makes the app barely usable since weeks. Hope this i prioritized in the next update. I have the problem on the work windows computer and also here on my private linux. Thank you!!

There are duplicates of many different types, which somehow all refer to the same problem: 
#10388 
#10219  
#10194

**Comment by pedr** [comment]
I spent some time trying to fix this, but I didn't make a lot of progress. What I gather:

- The behavior was introduced at `f19b1c536` with the Trash feature, looking at the changes introduced I couldn't find what might have been causing it.
- Before the introduction of the bug the note list would sort only the notes up until the note that was selected, moving the selected one to the top
- After the bug, the note list is sorting the notes after the select note, but the selected note is changed to the one that is occupying the position in the list the original selection sat in. (I could not find where this change happens, but the main suspect would be a faulty `key` property in the list)

**Comment by Dricc123** [comment]
Thanks @pedr for your efforts. In the issue description by @rcprcp here on the top of the page, it is mentioned another issue but "it may be a different bug". That other issue was solved so I wonder if it can give you a hint. Thanks again for looking into it.

**Comment by rcprcp** [comment]
This issue I have with using search, selecting an item from results, then typing into the body of the item, and it take the item out of the search result list and focusses a different item is still there on 3.0.8: 
<img width="262" alt="Screenshot 2024-05-22 at 18 26 51" src="https://github.com/laurent22/joplin/assets/17998205/29829c73-90f5-43dc-a6b6-1b8080784c22">

Thank you for checking into this.

**Comment by tessus** [comment]
Adding my info here (and closing my issue that was a duplicate of this one):

### Operating system

macOS

### Joplin version

3.0.8 (dev, darwin)

### Desktop version info

Joplin 3.0.8 (dev, darwin)

Client ID: e354a932e6e143c782264aad4112b674
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes

Revision: 50d08cd17 (dev)

Automatic Backlinks to note: 3.0.3
Backup: 1.4.1
Convert Text To New Note: 1.5.1
Get Notebook ID: 1.0.1
Insert Date: 1.0.1
Markdown table calculations: 1.0.5
Math Mode: 0.5.3
Menu items, Shortcuts, Toolbar icons: 1.1.0
Note list and sidebar toggle buttons: 1.0.3
Note overview: 1.6.0
Quick Links: 1.2.4
Table Formatter Plugin: 1.2.1
Templates: 2.3.0

### Current behaviour

1. enter a text in the global search box
2. click on a note that shows up in the note list
3. try to edit this note (click in the note and press a few keys)
4. Joplin switches to a random note and the note list changes as well


https://github.com/laurent22/joplin/assets/223439/e70d1981-c43c-4f81-9df2-98a52f74caa9


### Expected behaviour

I should be able to edit the note and the note list should not change.

### Logs

[log.txt](https://github.com/laurent22/joplin/files/15462318/log.txt)

**Comment by Dricc123** [comment]
Good to see this issue is on high priority! @alicehincu seems to be the person who solved a similar issue https://github.com/laurent22/joplin/pull/10370 maybe she can assist in solving this one ;-)

**Comment by personalizedrefrigerator** [comment]
This issue seems to be caused by the following two lines in f19b1c536439bc0a9f04253203371c15b60f65b3:

In `reducer.ts`,
```diff
-						} else if (isViewingAllNotes || noteIsInFolder(modNote, n.parent_id)) {
+						} else if (isViewingAllNotes || noteIsInFolder(modNote, draft.selectedFolderId)) {
```
[Full diff](https://github.com/laurent22/joplin/commit/f19b1c536439bc0a9f04253203371c15b60f65b3#diff-a05ff1dfe26df45f3dc8092366a4bea5e516a9c485524d5bbbcd5383368a667dL886-R914).

Reverting the change seems to fix both this issue and #10388, though this may cause trash-folder-related regressions.
**Edit**: @laurent22 Do you remember why this particular change was made?
   • **Edit 3**: This change allows right-click > "restore" to work in the trash folder.
**Edit 2**: Within the trash folder pull request, d63615b19c6610237aac0c796309b0e5dffd451c was the commit that specifically made this change.

**Comment by nonobio** [comment]
https://github.com/laurent22/joplin/issues/10388#issuecomment-2164597890
